bpf-common/src/program.rs

```diff
@@ -295,20 +297,27 @@ impl Program {
     }
     /// Poll a BPF_MAP_TYPE_HASH with a certain interval
     pub async fn poll<F, K, V>(
-        &self,
+        &mut self,
         map_name: &str,
         interval: Duration,
         mut poll_fn: F,
     ) -> Result<(), ProgramError>
     where
-        F: FnMut(Result<&mut HashMap<MapRefMut, K, V>, ProgramError>),
+        F: FnMut(Result<&mut HashMap<&mut MapData, K, V>, ProgramError>),
         F: Send + 'static,
         K: Pod + 'static + Send,
         V: Pod + 'static + Send,
     {
-        let mut map: HashMap<_, K, V> = self.bpf.map_mut(map_name)?.try_into()?;
+        let mut map_resource = self
+            .bpf
+            .take_map(map_name)
+            .ok_or_else(|| ProgramError::MapNotFound(map_name.to_string()))?;
         let mut rx_exit = self.tx_exit.subscribe();
         tokio::spawn(async move {
+            let mut map = match HashMap::try_from(&mut map_resource) {
+                Ok(map) => map,
+                Err(err) => return poll_fn(Err(err.into())),
+            };
```

Comment:
Check if it's possible to create the `Hashmap` outside of the task spawn. I saw this [https://docs.aya-rs.dev/user/aya/maps/struct.hashmap#method.try_from-2]() for owned map. in this way the signature of `poll_fn` could be simpler taking a plain `Hashmap` instead of the result